tensorboard/webapp/widgets/data_table/types.ts

```diff
@@ -43,6 +42,44 @@ export enum ColumnHeaderType {
   CUSTOM = 'CUSTOM',
 }
 
+export enum DomainType {
+  DISCRETE,
+  INTERVAL,
+}
+
+export type DiscreteFilterValues = string[] | number[] | boolean[];
+
+export type DiscreteFilterValue = DiscreteFilterValues[number];
+
+export interface DiscreteFilter {
+  type: DomainType.DISCRETE;
+  includeUndefined: boolean;
+  possibleValues: DiscreteFilterValues;
+  // Subset of `possibleValues`
+  filterValues: DiscreteFilterValues;
+}
+
+export interface DiscreteFilterChange {
+  hparamName: string;
+  includeUndefined: boolean;
+  filterValues: DiscreteFilterValues;
+}
+
+export interface IntervalFilter {
+  type: DomainType.INTERVAL;
+  includeUndefined: boolean;
+  minValue: number;
+  maxValue: number;
+  // Filter values have to be in between min and max values (inclusive).
+  filterLowerValue: number;
+  filterUpperValue: number;
+}
```

Comment:
These have all been copied over from `runs_data_source_types`. I actually prefer moving these all to an even more independent location but would prefer to do that as a later cleanup.